Cargo.lock

```diff
@@ -7049,6 +7050,7 @@ dependencies = [
  "solana-perf",
  "solana-program-runtime",
  "solana-rayon-threadlimit",
+ "solana-rpc-client",
```

Comment:
I appreciate the initiative on this change, and I know several other people have likely bumped into the issue this PR is trying to resolve (100% remember Firedancer bumping into this problem).

That being said, there has been some hesitation in the past to to pull RPC client into tools like this and ledger-tool. For example, hard-coding the RPC url's in `genesis_utils.rs` isn't the prettiest. It looks like the flags you added properly "do nothing" unless specified, so we won't break compatibility.

I wonder if there is some medium where we can add the CLI flag to activate / deactivate features AND also add a small wrapper script that calls `solana feature status` and parses output

Lastly, I think this might be a good one to either some feedback from others early on, maybe bring it up in Discord. I could be alone with my hesitation about adding `solana-rpc-client` as a dep, in which case, I don't want to add extra work for you on the front side

Reply:
yooo thank you for the heads up/comment. So as it turns out, I didn't push my last commit (updated now). But we don't actually need to pull the `solana-rpc-client` into `runtime`. We just need `solana-cli` in `runtime`. but yess will bring up in discord!